### ISSUE TITLE
Obtain token embeddings

### ISSUE THREAD

**Issue by IoSonoMarco** [mosaicml/llm-foundry#389]
Is there any straightforward way to simply obtain the token embeddings (last hidden state) from the HuggingFace or your repo implementations of mpt-7B? It seems I can only obtain logits for next work predictions

**Comment by sasaadi** [comment]
Maybe this:

```
from llmfoundry.models import MPTModel
model = MPTModel.from_pretrained(base_model_name, trust_remote_code=True, torch_dtype=torch.float16)
model.eval()
input_ids = tokenizer(your_text, max_length=256, return_tensors="pt")
output = model(input_ids=input_ids['input_ids'], output_hidden_states=True, return_dict=True)
token_embeddings = output["last_hidden_state"]

```

**Comment by alextrott16** [comment]
Yes, this recommendation is essentially correct, but I would suggest a slightly different route to leverage the AutoModel stuff:
```python
import torch
from transformers import AutoTokenizer, AutoModelForCausalLM

base_model_name = 'mosaicml/mpt-7b'
your_text = 'what a wonderful day to use llm foundry'
model = AutoModelForCausalLM.from_pretrained(base_model_name, trust_remote_code=True, torch_dtype=torch.bfloat16)
tokenizer = AutoTokenizer.from_pretrained(base_model_name, trust_remote_code=True)
input_ids = tokenizer(your_text, return_tensors="pt")
output = model(input_ids=input_ids['input_ids'], output_hidden_states=True, return_dict=True)
token_embeddings = output["hidden_state"][-1]
```
Or to use the base `MPTModel` as described above, replace the last 2 lines with (the result will be the same)
```python
output = model.transformer(input_ids=input_ids['input_ids'], output_hidden_states=True, return_dict=True)
token_embeddings = output["last_hidden_state"]
```
